tests/test_yaml_search_space/default_value_not_in_choices_config.yaml

```diff
@@ -1,4 +1,3 @@
-pipeline_space:
-  cat1:
-    choices: ["a", "b", "c"]
-    default: "d"
+cat1:
+  choices: ["a", "b", "c"]
+  default: "d"
```

Comment:
I appreciate your enthusiasm for testing, but this error should be caught by the `CategoricalParameter` during construction and it's less testing your yaml parsing and more testing the parameter construction.

I'm currently writing tests for Parameters and SearchSpace which would sort of duplicate a lot of properties tested.

Btw, not a bad thing, just letting you know that you don't have to test the whole system in your features! Leave some for the rest of us :)

Reply:
When revising the yaml usage, I will adjust the tests and remove those that affect the parameters